sdks/python/apache_beam/io/external/snowflake.py

```diff
@@ -0,0 +1,130 @@
+    :param schema: name of the Snowflake schema in the database to use.
+    :param database: name of the Snowflake database to use.
+    :param staging_bucket_name: name of the Google Cloud Storage bucket. Bucket will be used as a temporary location
+                           for storing CSV files. Those temporary directories will be named `sf_copy_csv_DATE_TIME_RANDOMSUFFIX`
+                           and they will be removed automatically once Read operation finishes.
+    :param storage_integration_name: is the name of a Snowflake storage integration object created according to [Snowflake documentation](https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html) for the GCS bucket.
+    :param csv_mapper: specifies a function which must translate user-defined object to array of strings.
+                  SnowflakeIO uses a [COPY INTO <location>](https://docs.snowflake.com/en/sql-reference/sql/copy-into-location.html)
+                  statement to move data from a Snowflake table to Google Cloud Storage as CSV files.
+                  These files are then downloaded via [FileIO](https://beam.apache.org/releases/javadoc/2.3.0/index.html?org/apache/beam/sdk/io/FileIO.html)
+                  and processed line by line. Each line is split into an array of Strings using the [OpenCSV](http://opencsv.sourceforge.net/) library.
+                  The csv_mapper function job is to give the user the possibility to convert the array of Strings
+                  to a user-defined type, ie. GenericRecord for Avro or Parquet files, or custom objects.
+                  Example:
+                   ```
+                        def csv_mapper(strings_array):
+ 		                    return User(strings_array[0], int(strings_array[1])))
+                   ```
+    :param table or query: specifies a Snowflake table name or custom SQL query
+    :param expansion_service: specifies URL of expansion service.
+
+    Authentication parameters:
+    It's required to pass one of the following combinations of valid parameters for authentication:
+    :param username and password: specifies username and password for username/password authentication method.
+    :param private_key_path and private_key_passphrase: specifies a private key file and password for key/ pair authentication method.
+    :param o_auth_token: specifies access token for OAuth authentication method.
+
+    """
+    self.params = ReadFromSnowflakeSchema(
+        server_name=server_name,
+        schema=schema,
+        database=database,
+        staging_bucket_name=staging_bucket_name,
+        storage_integration_name=storage_integration_name,
+        username=username,
+        password=password,
+        private_key_path=private_key_path,
+        private_key_passphrase=private_key_passphrase,
+        o_auth_token=o_auth_token,
+        table=table,
+        query=query
+    )
+    self.csv_mapper = csv_mapper
+    self.expansion_service = expansion_service
```

Comment:
We could use a default expansion_service like in Kafka

```
def default_io_expansion_service():
  return BeamJarExpansionService('sdks:java:io:expansion-service:shadowJar')

...

self.expansion_service = expansion_service or default_io_expansion_service()
```

Reply:
Thanks . done :)